firebase-database/src/main/java/com/google/firebase/database/core/utilities/PushIdGenerator.java

```diff
@@ -15,14 +15,22 @@
 package com.google.firebase.database.core.utilities;
 
 import static com.google.firebase.database.core.utilities.Utilities.hardAssert;
+import static com.google.firebase.database.core.utilities.Utilities.tryParseInt;
 
+import com.google.firebase.database.snapshot.ChildKey;
 import java.util.Random;
 
 public class PushIdGenerator {
 
   private static final String PUSH_CHARS =
       "-0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz";
 
+  private static final char MAX_PUSH_CHAR = 'z';
+
+  private static final char MIN_PUSH_CHAR = '-';
```

Comment:
Nit: Flip assignments for clarity. 

You could also refer to `PUSH_CHARS` and get the values from there (which will be slower and more code, so I am not sure it is actually worth it).

Reply:
I sort of prefer having separate sentinel values for this since they trigger edge-cases in the `successor`/`predecessor` code.